src/modules/add-site/components/pull-remote-site.tsx

```diff
@@ -118,9 +120,20 @@ export function PullRemoteSite( {
 	setSelectedRemoteSite: ( site?: SyncSite ) => void;
 } ) {
 	const { __ } = useI18n();
-	const { isAuthenticated } = useAuth();
+	const { isAuthenticated, user } = useAuth();
 	const { location } = useNavigator();
-	const { syncSites, refetchSites } = useSyncSites();
+
+	const { selectedSite } = useSiteDetails();
+	const { data: connectedSites = [] } = useGetConnectedSitesForLocalSiteQuery( {
+		localSiteId: selectedSite?.id,
+		userId: user?.id,
+	} );
+	const connectedSiteIds = connectedSites.map( ( { id } ) => id );
+	const { data: syncSites = [], refetch: refetchSites } = useGetWpComSitesQuery( {
+		connectedSiteIds,
+		userId: user?.id,
+	} );
```

Comment:
```suggestion
	const { data: syncSites = [] } = useGetWpComSitesQuery(
		{ connectedSiteIds, userId: user?.id },
		{ refetchOnMountOrArgChange: true }
	);
```
I believe we could remove the `refetchSites` call by passing the `refetchOnMountOrArgChange` flag here.

Reply:
OK, I will have a look at it, I had that initially, but it was refetching the sites more than expected. I will check it again after latest updates.

Reply:
I have done the changes as part of 61e8a4bc, they do the same refetch on mount and we avoid the `useEffect` 👍